unity-renderer/Assets/UIComponents/Scripts/Components/BaseComponentView.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using UnityEngine;
+
+public interface IBaseComponentView
+{
+    /// <summary>
+    /// It will be triggered after the UI component is initialized (it is called on the Start Unity event).
+    /// </summary>
+    event Action OnInitialized;
+
+    /// <summary>
+    /// Returns true if the UI component is already initialized.
+    /// </summary>
+    bool isInitialized { get; }
+
+    /// <summary>
+    /// It is called when the UI component game object awakes.
+    /// </summary>
+    void Initialize();
+
+    /// <summary>
+    /// Shows the UI component.
+    /// </summary>
+    /// <param name="instant">True for not apply progressive animation.</param>
+    void Show(bool instant = false);
+
+    /// <summary>
+    /// Hides the UI component.
+    /// </summary>
+    /// <param name="instant">True for not apply progressive animation.</param>
+    void Hide(bool instant = false);
+
+    /// <summary>
+    /// Updates the UI component with the current model.
+    /// </summary>
+    void RefreshControl();
+
+    /// <summary>
+    /// It is called when the UI component game object is destroyed.
+    /// </summary>
+    void Dispose();
+}
+
+[RequireComponent(typeof(ShowHideAnimator))]
+[RequireComponent(typeof(CanvasGroup))]
```

Comment:
@BrianAmadori I'm a bit concerned about performance/design here. 

It's enforcing a CanvasGroup/ShowHideAnimator wise? 
For some UIElements that don't require a fade transition a simple `SetActive` will be more performant and for other UIElements maybe we want to use a tween or any other kind of transition for Show/Hide.

 We could have a couple of interfaces `IUIShow` `IUIHide` and through composition and RequireComponent works with interfaces have some:
`[RequireComponent(typeof(IUIShow))]`
`[RequireComponent(typeof(IUIHide))]`

This would add lot of flexibility. You can have different effects for showing and hiding or even rely on an animator to do crazy stuff, manage particle effects, etc...

Reply:
I think it's a good point what you say, @AjimenezDCL. We shouldn't forze to all UI components to use an animator for the show/hide.

My suggestion for this is simply to remove the `[RequireComponent]` tags and let the child classes to override (or not) the Show/Hide methods. If they are not overriden, the UI component will try to use the default `ShowHideAnimator` feature as usual. And if they are overriden, the UI component will be able to use any other custom way of visibility.

Anyway let wait for what @BrianAmadori thinks about it.

Reply:
Yeah, the solution proposed by @sandrade-dcl looks fair enough to me. Anyway let's remember that performance is not an issue with UI, and even so, we are not going to run an animation constantly, just for showing/hiding.